tests/modal.ts

```diff
@@ -0,0 +1,1097 @@
+                                           width:20px;margin:auto;
+                                           border-top:3px solid #000">
+                               </div>
+                            </div>
+                            <div class="left-guide" style="background:blue;
+                                                           position:static">
+                                <p>class: left-guide, pos: left</p>
+                                <div style="background:green;display:flex;
+                                            position:sticky;left:0px;
+                                            height:20px;width:20px;
+                                            border-left:4px solid #000;
+                                            margin:auto;">
+                                </div>
+                            </div>
+                            <div class="bottom-guide" style="background:red">
+                                <p> class: bottom-guide, pos: bottom</p>
+                                <div style="background:grey;height:20px;
+                                            width:20px;margin:auto;
+                                            border-top:3px solid #000">
+                                </div>
+                            </div>
+                            <div style="height:200px;width:250px;
+                                        background:green">
+                                <h1>h1</h1>
+                            </div>
+                            <div style="height:300px;width:200px;
+                                        background:yellow">
+                                <h2>h2</h2>
+                            </div>
+                            <div style="height:350px;width:250px;
+                                        background:blue">
+                                <h3 style="height:35px;width:25px">h3</h3>
+                            </div>
+                        </div>
+                    </aalam-modal>
+                </div>
+            </div>
+       `);
+       const modal = el1.querySelector("#modal");
+       const modal_body = el1.querySelector("#modal-body");
+       const r_guide = modal.querySelector(".right-guide");
+       const l_guide = modal.querySelector(".left-guide");
+       const t_guide = modal.querySelector(".top-guide");
+       const b_guide = modal.querySelector(".bottom-guide");
+       modal.animationDur = 0;
+       modal_body.style.height="100%";
+       modal_body.style.overflow="auto";
+       modal_body.style.margin="auto";
+       modal_body.style.boxSizing="border-box";
+       modal.show();
+       await modal.updated();
+       expect(modal.parentElement).to.equal(document.body);
+
+       await setViewport({width:1400, height:640});
+       let bounds = modal_body.getBoundingClientRect();
+       await sendMouse({type:'move', position: [1300, 50]});
```

Comment:
With this hardcoded numbers its difficult to identify which guide is getting dragged.